src/_pytest/terminal.py

```diff
@@ -597,6 +598,16 @@ def pytest_report_header(self, config):
         plugininfo = config.pluginmanager.list_plugin_distinfo()
         if plugininfo:
             result.append("plugins: %s" % ", ".join(_plugin_nameversions(plugininfo)))
+
+        if config._implicit_args:
+            result.append(
+                "implicit args: %s"
+                % ", ".join(
+                    "%r (%s)" % (" ".join([shell_quote(x) for x in values]), desc)
```

Comment:
I suggest to break this expression in some variables, as it is formatted it is quite hard to parse.

Reply:
Perhaps we can have one section per line? For example at work we set `PYTEST_ADDOPTS=--max-slave-restart=10 --timeout=60 -ra --color=yes --replay-record-dir=build/tests/replay`, which can get quite big.